packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -296,8 +303,11 @@ class ButtonBase extends React.Component {
         onTouchEnd={this.handleTouchEnd}
         onTouchMove={this.handleTouchMove}
         onTouchStart={this.handleTouchStart}
-        onContextMenu={this.handleContextMenu}
-        ref={innerRef || buttonRef}
+        ref={ref => {
```

Comment:
What about creating an instance method not to call setRef every time the ButtonBase renders?

Reply:
Got the same idea when I read the review below :smile: Will investigate if we can configure a lint rule to check for this.

Reply:
This wouldn't handle change of props. This behavior is however required for the SpeedDial. Since it's an optimization we can tackle this in a different PR.

Reply:
Yeah sure.